python/pip_install/requirements.bzl

```diff
@@ -0,0 +1,76 @@
+"Rules to verify and update pip-compile locked requirements.txt"
+
+load("//python:defs.bzl", "py_binary", "py_test")
+load("//python/pip_install:repositories.bzl", "requirement")
+
+def compile_pip_requirements(
+        name,
+        extra_args = [],
+        visibility = ["//visibility:private"],
+        requirements_in = "requirements.in",
+        requirements_txt = "requirements.txt",
+        **kwargs):
+    """
+    Produce two targets for checking pip-compile:
+
+    - validate with `bazel test <name>_test`
+    - update with   `bazel run <name>.update`
+
+    Args:
+        name: base name for generated targets
+        extra_args: passed to pip-compile
+        visibility: passed to both the _test and .update rules
+        requirements_in: file expressing desired dependencies
+        requirements_txt: result of "compiling" the requirements.in file
+        **kwargs: other bazel attributes passed to the "_test" rule
+    """
+    requirements_in = kwargs.pop("requirements_in", name + ".in")
+    requirements_txt = kwargs.pop("requirements_locked", name + ".txt")
+
+    data = kwargs.pop("data", []) + [requirements_in, requirements_txt]
+
+    loc = "$(rootpath %s)"
+
+    # Use the Label constructor so this is expanded in the context of the file
+    # where it appears, which is to say, in @rules_python
+    pip_compile = Label("//python/pip_install:pip_compile.py")
+
+    args = [
+        loc % requirements_in,
+        loc % requirements_txt,
+        name + ".update",
+    ] + extra_args
+
+    deps = [
+        requirement("click"),
```

Comment:
`click` seems to be a transitive dep? Although I don't see it in the locked requirements

Reply:
ah, there are two different requirements lists

the one in requirements.in/requirements.txt is just being refactored to use the new rule, to prove that it works

The rule itself gets bootstrapped by the list in repositories.bzl which is also how users get the dependencies. `click` is in there.